File: server/src/modules/chat/voice-plain.util.ts
/**
 * 从助手 Markdown 派生适合 TTS / 语音字幕的纯文本（与 Flutter 侧 fallback 逻辑保持大致一致）。
 */
export function stripMarkdownForVoice(markdown: string): string {
  let t = markdown.trim();
  if (!t) return '';

  t = t.replace(/```[\s\S]*?```/g, ' ');
  t = t.replace(/`[^`\n]+`/g, ' ');
  t = t.replace(/^#{1,6}\s+/gm, '');
  t = t.replace(/\*\*([^*]+)\*\*/g, '$1');
  t = t.replace(/\*([^*]+)\*/g, '$1');
  t = t.replace(/__([^_]+)__/g, '$1');
  t = t.replace(/_([^_]+)_/g, '$1');
  t = t.replace(/\[([^\]]+)\]\([^)]+\)/g, '$1');
  t = t.replace(/^>\s?/gm, '');
  t = t.replace(/^\s*[-*+]\s+/gm, '');
  t = t.replace(/^\s*\d+\.\s+/gm, '');
  t = t.replace(/\n{3,}/g, '\n\n');

  return t.replace(/\s+/g, ' ').trim();
}

File: app/lib/features/chat/presentation/voice_plain_util.dart
import 'package:starpath/features/chat/domain/chat_model.dart';

/// 与 Nest [stripMarkdownForVoice] 保持大致一致，供本地 TTS / 字幕 fallback。
String stripMarkdownForVoice(String markdown) {
  var t = markdown.trim();
  if (t.isEmpty) return '';

  t = t.replaceAll(RegExp(r'```[\s\S]*?```'), ' ');
  t = t.replaceAll(RegExp(r'`[^`\n]+`'), ' ');
  t = t.replaceAll(RegExp(r'^#{1,6}\s+', multiLine: true), '');
  t = t.replaceAllMapped(RegExp(r'\*\*([^*]+)\*\*'), (m) => m[1]!);
  t = t.replaceAllMapped(RegExp(r'\*([^*]+)\*'), (m) => m[1]!);
  t = t.replaceAllMapped(RegExp(r'__([^_]+)__'), (m) => m[1]!);
  t = t.replaceAllMapped(RegExp(r'_([^_]+)_'), (m) => m[1]!);
  t = t.replaceAllMapped(RegExp(r'\[([^\]]+)\]\([^)]+\)'), (m) => m[1]!);
  t = t.replaceAll(RegExp(r'^>\s?', multiLine: true), '');
  t = t.replaceAll(RegExp(r'^\s*[-*+]\s+', multiLine: true), '');
  t = t.replaceAll(RegExp(r'^\s*\d+\.\s+', multiLine: true), '');
  t = t.replaceAll(RegExp(r'\n{3,}'), '\n\n');

  return t.replaceAll(RegExp(r'\s+'), ' ').trim();
}

/// TTS / 语音字幕：优先服务端派生字段，否则本地 strip。
String voicePlainForMessage(MessageModel m) {
  final v = m.voicePlain?.trim();
  if (v != null && v.isNotEmpty) return v;
  return stripMarkdownForVoice(m.content);
}
